python_modules/dagster/dagster/_core/definitions/result.py

```diff
@@ -0,0 +1,31 @@
+from typing import NamedTuple, Optional
+
+from .events import (
+    AssetKey,
+    CoercibleToAssetKey,
+)
+from .metadata import MetadataUserInput
+
+
+class MaterializeResult(
+    NamedTuple(
+        "_MaterializeResult",
+        [
+            ("asset_key", Optional[AssetKey]),
+            ("metadata", Optional[MetadataUserInput]),
+        ],
+    )
+):
```

Comment:
is there a reason why we are still using this NamedTuple format instead of new style

```
class MaterializeResult(NamedTuple):
  asset_key: Optional[AssetKey]
  metadata: Optional[MetadataUserInput]
```

Reply:
when you override `__new__` (in this case to coerce asset key) you need to do the old style

Reply:
devastating